### DISCUSSION TITLE
Possible to rollback session if a transaction fail

### DISCUSSION THREAD

**Discussion by kennywongck** [fastapi/sqlmodel#1034] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# code to save record to database
		def save_records(session: SessionDep, unique_id: str, str_to_insert: str):
			record_to_insert = RecordModel(
				unique_id = unique_id
				str_to_insert = str_to_insert
			)
			session.add(record_to_insert)
			session.commit()
			session.refresh(record_to_insert)
			return record_to_insert
```

### Description

I try to save records in my fastapi application with the code snippet above, but when my application received a lot of concurrent request, looks like one of the session fail or hang and I got the error below which cause the whole database to fail

`sqlalchemy.exc.PendingRollbackError: Can't reconnect until invalid transaction is rolled back.  Please rollback() fully before proceeding (Background on this error at: https://sqlalche.me/e/20/8s2b)
`
Is it possible to do a session.rollback() in case a transaction fail or it is handle by sqlmodel already internally?

### Operating System

Linux

### SQLModel Version

sqlmodel==0.0.16

### Python Version

python 3.11

**Comment by DSLituiev** [comment]
This trick has worked for me:

```python
while session.in_transaction():
    session.rollback()
```

**Comment by YuriiMotov** [comment]
I think this should already work this way if you use context manager to open session
